server/lib/s3backup/__init__.py

```diff
@@ -246,6 +267,30 @@ def __init__(self, config):
                                          aws_secret_access_key='{}'.format(
                                              self.secret_access_key),
                                          endpoint_url='{}'.format(self.endpoint_url))
+        self.logger = logger
+
+    @staticmethod
+    def calculate_multipart_etag(tb, chunk_size):
+        md5s = []
+
+        with open(tb, 'rb') as fp:
+            while True:
+                data = fp.read(chunk_size)
+                if not data:
+                    break
+                md5s.append(hashlib.md5(data))
+
+        if len(md5s) > 1:
+            digests = b"".join(m.digest() for m in md5s)
+            new_md5 = hashlib.md5(digests)
+            new_etag = '{}-{}'.format(new_md5.hexdigest(), len(md5s))
+        elif len(md5s) == 1:
+            # file smaller than chunk size
+            new_etag = '{}'.format(md5s[0].hexdigest())
```

Comment:
Isn't a `.hexdigest()` already a string object?  So this could be written as:

```
new_etag = md5s[0].hexdigest()
```

Reply:
Yes.